woocommerce/woocommerce-android
# [Simplified login] Enhance WPComWebView for jetpack connection usage

Closes: #7797 


⚠️ Please don't merge before #7889 

### Description
This PR adds the following changes:
1. Converts the WPComWebView to Compose, this brings some advantages as mentioned below.
2. Allow passing multiple URLs to trigger exit from the WebView, this is needed for the Jetpack Connection flow.
3. Allow showing a cross (modal display) in the Toolbar.
4. Decouple `WPComWebView` from MainActivity: it uses its Toolbar and doesn't use the BackListener.
    This is needed if we want to launch this fragment from the LoginActivity in the future.
 
As mentioned above, the conversion to Compose brings an advantage, as the `WCWebView` component we have supports capturing back button presses, now when the user presses the back button and they have navigated to deep destinations in the WebView, we will go to the previous page instead of closing the fragment (cc @kidinov @AnirudhBhat this solves the issue mentioned [here](https://github.com/woocommerce/woocommerce-android/pull/7355#issuecomment-1239039917)).
The navigation up (using the Toolbar's navigation icon) still closes the Fragment, this aligns with how custom chrome tabs behave, and we can still customize it if needed in the Future.

### Testing instructions
Please test some scenarios that use the authenticated webview, and confirm there is no regression, some flows:

- Card reader purchase that can be accessed through the Payments screen (accessible from the More screen).
- Woo installation, this can be tested using the following steps:
    1. Uninstall Woo from one of your sites.
    2. Open the site picker.
    3. Click on "Add a store"
    4. Click on "Connect existing store".
    5. Enter the URL of the non-woo store.
    6. Click "Install Woo", and confirm the flow still works.
- Jetpack Connection: this can be tested by following the steps of #7889.

### Images/gif
For Jetpack Activation, the flow looks like this now (as modal):

https://user-images.githubusercontent.com/1657201/203388198-39d8225a-4346-43ba-8647-86b9ce5fc921.mp4
